samples/secure_tunneling/secure_tunnel/main.cpp

```diff
@@ -0,0 +1,278 @@
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+#include <aws/crt/Api.h>
+#include <aws/iot/MqttClient.h>
+#include <aws/iotdevicecommon/IotDevice.h>
+#include <aws/iotsecuretunneling/SecureTunnel.h>
+
+#include <fstream>
+
+using namespace std;
+using namespace Aws::Crt;
+using namespace Aws::Iotsecuretunneling;
+using namespace Aws::Crt::Io;
+
+static void s_printHelp()
+{
+    fprintf(stdout, "Usage:\n");
+    fprintf(
+        stdout,
+        "secure-tunnel\n"
+        "--region                 <region>\n"
+        "--ca_file                <optional: path to custom ca>\n"
+        "--access_token_file      <path to access token> or "
+        "--access_token           <access token>\n"
+        "--localProxyModeSource   <optional: sets to Source Mode>\n"
+        "--message                <optional: message to send>\n\n");
+    fprintf(stdout, "region: the region of your iot thing and secure tunnel\n");
+    fprintf(
+        stdout,
+        "ca_file: Optional, if the mqtt server uses a certificate that's not already"
+        " in your trust store, set this.\n");
+    fprintf(stdout, "\tIt's the path to a CA file in PEM format\n");
+    fprintf(stdout, "access_token_file: path to the tunneling access token file\n");
+    fprintf(stdout, "access_token: tunneling access token\n");
+    fprintf(stdout, "localProxyModeSource: Use to set local proxy mode to source. Default is destination\n");
+    fprintf(stdout, "message: message to send. Default: 'Hello World'\n");
+}
+
+bool s_cmdOptionExists(char **begin, char **end, const String &option)
+{
+    return std::find(begin, end, option) != end;
+}
+
+char *s_getCmdOption(char **begin, char **end, const String &option)
+{
+    char **itr = std::find(begin, end, option);
+    if (itr != end && ++itr != end)
+    {
+        return *itr;
+    }
+    return 0;
+}
+
+int main(int argc, char *argv[])
+{
+    ApiHandle apiHandle;
+
+    String region;
+    String endpoint;
+    String caFile;
+    String accessToken;
+    String message = "Hello World";
```

Comment:
trivial: weird mix of `Aws::Crt::String` and `std::string`
maybe just stick with std::string because this is a sample, and they shouldn't be using our weird internal typedefs